website/views.py

```diff
@@ -17,27 +17,52 @@
 from django.core.mail import EmailMultiAlternatives
 from .spamFilter import predict, train
 from .decorators import check_recaptcha
+from django.urls import resolve, Resolver404
+import openpyxl
 
 User = get_user_model()
 admins = (
-   9, 4376, 4915, 14595, 12329, 22467, 5518, 30705
+    9, 4376, 4915, 14595, 12329, 22467, 5518, 30705
 )
-categories = FossCategory.objects.order_by('name')
 
 # Function to check if user is in any moderator group
-def is_moderator(user):
-    return user.groups.count() > 0
 
-# Function to check if user is anonymous, and if not he/she has first_name and last_name
+
+
+def is_moderator(user,question=None):
+    if question:
+        if user.groups.filter(moderatorgroup=ModeratorGroup.objects.get(category=question.category)).exists():
+            return True
+    elif user.groups.count() > 0:
```

Comment:
Why only ```user.groups.count() > 0``` is not enough to say user is moderator?
Why ```user.groups.filter(moderatorgroup=.....``` is needed?

Reply:
while editing the question or its answer or comments, moderator should be checked whether he is belong to the moderator group of the question's category or not
Otherwise User of any moderator group of specific category can edit the questions of any other categories